manifests/base.pp

```diff
@@ -18,13 +18,17 @@
         }
       }
       'RedHat': {
-        yumrepo { 'upstream_rsyslog':
-          ensure   => 'present',
-          descr    => 'Adiscon Enterprise Linux rsyslog',
-          baseurl  => 'http://rpms.adiscon.com/v8-stable/epel-$releasever/$basearch',
-          enabled  => '1',
-          gpgcheck => '0',
-          gpgkey   => 'http://rpms.adiscon.com/v8-stable/epel-$releasever/$basearch',
+        if $facts['os']['release']['major'] == 8 {
+          fail('Upstream Repo does not currently work with EL8!')
+        } else {
```

Comment:
Nit: you don't need the `} else {` and you can simply fail. That makes the diff smaller.

Reply:
You'd think I'd realize that since I write guards without else all the time in Go and Ruby :man_shrugging: